packages/notebook-extension/src/index.tsx

```diff
@@ -1060,6 +1065,87 @@ const activeCellTool: JupyterFrontEndPlugin<void> = {
   }
 };
 
+export const filterCellsPlugin: JupyterFrontEndPlugin<void> = {
+  id: '@jupyterlab/notebook-extension:filter-cells',
+  description: 'Filter cells.',
+  autoStart: true,
+  requires: [INotebookTracker, IFormRendererRegistry],
+  optional: [ICommandPalette, ITranslator],
+  activate: (
+    app: JupyterFrontEnd,
+    tracker: INotebookTracker,
+    formRegistry: IFormRendererRegistry,
+    palette: ICommandPalette | null,
+    translator_: ITranslator | null
+  ): void => {
+    const translator = translator_ ?? nullTranslator;
+    const trans = translator.load('jupyterlab');
+    const { shell, commands } = app;
+    const isEnabled = () => Private.isEnabled(shell, tracker);
+    const notebookModelMap = new WeakMap<Notebook, CellFiltersModel>();
+
+    const filterButton: IFormRenderer = {
+      fieldRenderer: () => (
+        <CommandToolbarButtonComponent
+          commands={commands}
+          id={CommandIDs.filterCells}
+          caption={trans.__('Open the filtering tool')}
+        ></CommandToolbarButtonComponent>
+      )
+    };
+    formRegistry.addRenderer(
+      '@jupyterlab/notebook-extension:filter-cells.renderer',
+      filterButton
+    );
+
+    app.commands.addCommand(CommandIDs.filterCells, {
+      label: trans.__('Filter Cells'),
+      caption: trans.__('Filter cells'),
+      execute: async args => {
+        const current = getCurrent(tracker, shell, args);
+
+        if (current) {
+          let model = notebookModelMap.get(current.content);
+          if (!model) {
+            model = new CellFiltersModel({ notebook: current.content });
+            notebookModelMap.set(current.content, model);
+          }
+
+          const selectButton = Dialog.okButton({
+            label: trans.__('Filter Cell(s)'),
+            actions: ['select']
+          });
+          const clearButton = Dialog.warnButton({
+            label: trans.__('Clear Filter(s)'),
+            actions: ['clear']
+          });
+          const view = new CellFiltersView({ model, translator });
+          const result = await showDialog({
+            title: trans.__('Select cell filters:'),
+            body: view,
+            buttons: [Dialog.cancelButton(), selectButton, clearButton]
```

Comment:
`clearButton` (destructive action) is the default; should it be changed so that cancel is the default? I can see the case for either, feel free to ignore if this was a deliberate decision.

Reply:
Sorry but how do you define or know that the `clearButton` is the default one ?

Reply:
I have put the cancel button as the last element of the list of buttons of the dialog.

Reply:
[`Dialog.IOptions` has `defaultButton`](https://jupyterlab.readthedocs.io/en/latest/api/interfaces/apputils.Dialog.IOptions.html#defaultButton) option for this purpose so maybe you could restore the order as it was before?